### PR TITLE
Set up of redux store, actions, root reducer and implemented the theme toggling there.

### PR DESCRIPTION
**Pull request by KZzizzle**
So far, the only action available is the theme toggling. However, file structure is in place for you to add what you want. All redux stuff is in the new directory aptly named "redux_stuff"

### CODE REVIEW

File: src/redux_stuff/actions.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {  TOGGLE_THEME } from "./constants";
```
Comment (ignapas) [inline src/redux_stuff/actions.js]:
```suggestion
import { TOGGLE_THEME } from "./constants";
```

File: src/menu/Menu.js [hunk 2]
```diff
@@ -27,29 +38,31 @@ const NavigationButton = props => {
...
-      <IconButton onClick={props.toggleDarkModeHandler}>
-        { theme.palette.type === 'dark' ? (
-          <Brightness7 />
-        ) : (
-          <Brightness4 />
-        )}
-      </IconButton> 
-    </Tooltip>
-  );
+      <Box className="dark-mode-toggle-container">
```
Comment (ignapas) [inline src/menu/Menu.js]: Is this Box necessary?

File: src/redux_stuff/actions.js (hunk 1 above)
Comment (KZzizzle) [reply src/redux_stuff/actions.js]: removed space

File: src/menu/Menu.js (hunk 2 above)
Comment (KZzizzle) [reply src/menu/Menu.js]: nope, took it out

**Pull request merged**